src/test/java/bio/terra/integration/AccessTest.java

```diff
@@ -142,27 +155,132 @@ public void checkShared() throws  Exception {
                     startsWith("Access Denied:"));
+                return false;
             }
-        }
+        });
+
 
         assertThat("reader can access the dataset after it has been shared",
             hasAccess,
             equalTo(true));
-
     }
 
+    @Test
+    public void fileAclTest() throws Exception {
+        BigQueryProject bigQueryProject = getBigQueryProject(readerToken);
+
+        studySummaryModel = dataRepoFixtures.createStudy(steward, "file-acl-test-study.json");
+        String gsPath = "gs://" + testConfiguration.getIngestbucket();
+
+        FSObjectModel fsObjectModel = dataRepoFixtures.ingestFile(
+            steward,
+            studySummaryModel.getId(),
+            gsPath + "/files/File%20Design%20Notes.pdf",
+            "/foo/bar");
+
+        String json = String.format("{\"file_id\":\"foo\",\"file_ref\":\"%s\"}", fsObjectModel.getObjectId());
+
+        String targetPath = "scratch/file" + UUID.randomUUID().toString() + ".json";
+        BlobInfo targetBlobInfo = BlobInfo
+            .newBuilder(BlobId.of(testConfiguration.getIngestbucket(), targetPath))
+            .build();
+
+        try (WriteChannel writer = storage.writer(targetBlobInfo)) {
+            writer.write(ByteBuffer.wrap(json.getBytes("UTF-8")));
+        }
+
+        IngestResponseModel ingestResponseModel = dataRepoFixtures.ingestJsonData(
+            steward,
+            studySummaryModel.getId(),
+            "file",
+            targetPath);
+
+        assertThat("1 Row was ingested", ingestResponseModel.getRowCount(), equalTo(1L));
+
+        DatasetSummaryModel datasetSummaryModel = dataRepoFixtures.createDataset(
+            custodian,
+            studySummaryModel,
+            "file-acl-test-dataset.json");
+
+        dataRepoFixtures.addDatasetPolicyMember(
+            custodian,
+            datasetSummaryModel.getId(),
+            SamClientService.DataRepoRole.READER,
+            reader.getEmail());
+
+        AuthenticatedUserRequest authenticatedReaderRequest =
+            new AuthenticatedUserRequest(reader.getEmail(), readerToken);
+        assertThat("correctly added reader", samClientService.isAuthorized(
+            authenticatedReaderRequest,
+            SamClientService.ResourceType.DATASET,
+            datasetSummaryModel.getId(),
+            SamClientService.DataRepoAction.READ_DATA), equalTo(true));
+
+        TestUtils.flappyExpect(5, samTimeout, true, () -> {
+            try {
+                boolean datasetExists = bigQueryProject.datasetExists(datasetSummaryModel.getName());
+                assertThat("Dataset wasn't created right", datasetExists, equalTo(true));
+                return true;
+            } catch (PdaoException e) {
+                assertThat(
+                    "checking message for pdao exception error",
+                    e.getCause().getMessage(),
+                    startsWith("Access Denied:"));
+                return false;
+            }
+        });
```

Comment:
this code (lambda to flappyExpect) is exactly the same as in the test above, but we are not checking the result. is this just so that we wait until it is accessible? maybe this can be simplified or if we do want the exact same code should we make it a private method?